subsys/net/ip/tcp2.c

```diff
@@ -1129,20 +1129,19 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		 * ACK , shouldn't we go to SYN RECEIVED state? See Figure
 		 * 6 of RFC 793
 		 */
-		if (FL(&fl, &, ACK, th && th_ack(th) == conn->seq)) {
+		if (FL(&fl, &, SYN | ACK, th && th_ack(th) == conn->seq)) {
 			tcp_send_timer_cancel(conn);
-			next = TCP_ESTABLISHED;
-			net_context_set_state(conn->context,
-					      NET_CONTEXT_CONNECTED);
-			if (FL(&fl, &, PSH)) {
+			conn_ack(conn, th_seq(th) + 1);
+			if (len) {
```

Comment:
have an empty line before the if (since previous line it not doing anything with len)